components/slack/sources/common/base.mjs

```diff
@@ -0,0 +1,84 @@
+import slack from "../../slack.app.mjs";
+
+export default {
+  props: {
+    slack,
+    nameCache: "$.service.db",
+  },
+  methods: {
+    ...slack.methods,
+    async maybeCached(key, refreshVal, timeoutMs = 3600000) {
+      let record = this.nameCache.get(key);
+      const time = Date.now();
+      if (!record || time - record.ts > timeoutMs) {
+        record = {
+          ts: time,
+          val: await refreshVal(),
+        };
+        this.nameCache.set(key, record);
+      }
+      return record.val;
+    },
+    async getUserName(id) {
+      return this.maybeCached(`users:${id}`, async () => {
+        const info = await this.slack.sdk().users.info({
+          user: id,
+        });
+        if (!info.ok) throw new Error(info.error);
+        return info.user.name;
+      });
+    },
+    async getBotName(id) {
+      return this.maybeCached(`bots:${id}`, async () => {
+        const info = await this.slack.sdk().bots.info({
+          bot: id,
+        });
+        if (!info.ok) throw new Error(info.error);
+        return info.bot.name;
+      });
+    },
+    async getConversationName(id) {
+      return this.maybeCached(`conversations:${id}`, async () => {
+        const info = await this.slack.sdk().conversations.info({
+          channel: id,
+        });
+        if (!info.ok) throw new Error(info.error);
+        if (info.channel.is_im) {
+          return `DM with ${await this.getUserName(info.channel.user)}`;
+        } else {
+          return info.channel.name;
+        }
+      });
+    },
+    async getTeamName(id) {
+      return this.maybeCached(`team:${id}`, async () => {
+        try {
+          const info = await this.slack.sdk().team.info({
+            team: id,
+          });
+          return info.team.name;
+        } catch (err) {
+          console.log("Error getting team name, probably need to re-connect the account at pipedream.com/apps", err);
+          return id;
+        }
+      });
+    },
+    processEvent(event) {
+      return event;
+    },
+  },
+  async run(event) {
+    event = await this.processEvent(event);
+
+    if (event) {
+      if (!event.client_msg_id) {
+        event.pipedream_msg_id = `pd_${Date.now()}_${Math.random().toString(36)
+          .substr(2, 10)}`;
+      }
+
+      this.$emit(event, {
+        id: event.client_msg_id || event.pipedream_msg_id,
+      });
```

Comment:
Need to emit a summary and a timestamp. If the summaries/timestamps should be different for different components, you might want to create `generateMeta` methods for them.
```suggestion
      this.$emit(event, {
        id: event.client_msg_id || event.pipedream_msg_id,
        summary: "Some message here",
        ts: Date.now() // if available, set the timestamp to the date the event occurred returned in the event.
      });
```